rest/config_test.go

```diff
@@ -518,25 +518,77 @@ func TestMergeWith(t *testing.T) {
 
 func TestDeprecatedConfigLoggingFallback(t *testing.T) {
 	defer base.SetUpTestLogging(base.LevelInfo, base.KeyAll)()
-	logFilePath := "/var/log/sync_gateway"
-	dlfPath := "/tmp/log/sync_gateway"
 	logKeys := []string{"Admin", "Access", "Auth", "Bucket", "Cache"}
 	deprecatedLog := []string{"Admin", "Access", "Auth", "Bucket", "Cache"}
 
-	ddl := &base.LogAppenderConfig{LogFilePath: &logFilePath, LogKeys: logKeys, LogLevel: base.PanicLevel}
-	lc := &base.LoggingConfig{DeprecatedDefaultLog: ddl}
-	sc := &ServerConfig{Logging: lc, DeprecatedLogFilePath: &dlfPath, DeprecatedLog: deprecatedLog}
+	// Create temp dirs for specifying logging config
+	logFilePath, err := ioutil.TempDir("", "logFilePath")
+	require.NoErrorf(t, err, "Error creating temp dir %q", logFilePath)
+	dlfPath, err := ioutil.TempDir("", "dlfPath")
+	require.NoErrorf(t, err, "Error creating temp dir %q", dlfPath)
 
-	warns := sc.deprecatedConfigLoggingFallback()
-	assert.Equal(t, filepath.Dir(*sc.Logging.DeprecatedDefaultLog.LogFilePath), sc.Logging.LogFilePath)
+	defer func() {
+		require.NoErrorf(t, os.RemoveAll(logFilePath), "Error removing dir %q", logFilePath)
+		require.NoErrorf(t, os.RemoveAll(dlfPath), "Error removing dir %q", dlfPath)
+		_, err := os.Stat(logFilePath)
+		require.True(t, os.IsNotExist(err), "Deleted dir %q shouldn't exist", logFilePath)
+		_, err = os.Stat(logFilePath)
+		require.True(t, os.IsNotExist(err), "Deleted dir %q shouldn't exist", dlfPath)
+	}()
+
+	// Call deprecatedConfigLoggingFallback with specifying deprecated default log file path as a directory
+	ddl := &base.LogAppenderConfig{
+		LogFilePath: &logFilePath,
+		LogKeys:     logKeys,
+		LogLevel:    base.PanicLevel,
+	}
+	lc := &base.LoggingConfig{
+		DeprecatedDefaultLog: ddl,
+	}
+	sc := &ServerConfig{
+		Logging:               lc,
+		DeprecatedLogFilePath: &dlfPath,
+		DeprecatedLog:         deprecatedLog,
+	}
+
+	warns, err := sc.deprecatedConfigLoggingFallback()
```

Comment:
This seems like a perfect use case for iterating over sub-tests - as written, it's hard to identify exactly what config options are being tested.